feat: Swaps tron integration

bfullam opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Adds the Tron network to swaps/bridge.

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: added Tron network support in swaps

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds Tron support to swaps and bridge, filters non‑tradable Tron tokens, updates fee/status handling, defaults, and dependencies.
> 
> - **Bridge/Swaps Integration**
>   - Allow `TrxScope.Mainnet` in swaps (`isSwapsAllowed`) and bridge (`ALLOWED_BRIDGE_CHAIN_IDS`, network name map).
>   - Add Tron default swap token (`USDT` TRC20) in `default-swap-dest-tokens.ts` and BIP44 pairs.
> - **Token Handling**
>   - New `isTradableToken` utility to exclude Tron resource tokens (`Energy`, `Bandwidth`, `Max Bandwidth`).
>   - `useTokens` now filters non‑tradable tokens, normalizes non‑EVM addresses, and improves deduping/exclusions.
> - **UI/Logic**
>   - `TransactionDetails`: safer multichain fee sum (handles partially fungible fees) and unified status selection.
>   - Asset utils: mark Solana/Tron assets as swaps‑allowed.
> - **Tests**
>   - Extensive tests for non‑EVM normalization, Tron token filtering, deduplication, and edge cases.
> - **Dependencies**
>   - Bump `@metamask/bridge-controller` and `@metamask/bridge-status-controller` to `^60.1.0`; `@metamask/tron-wallet-snap` to `^1.7.x` (lockfile updated).
> - **Docs**
>   - Minor README fixes (overlay usage, sample E2E notes).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b88a55b7a2f4015d1a1f8c14ba24755ba41e97a5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

bfullam committed:
chore: check for tron in signing check

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
chore: add Tron to short network name map

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
chore: allow tron assets for swap

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
chore: bump tron snap

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
fix: remove duplicate import

georgewrmarshall reviewed (dismissed):
Approving lint fix on component library file to unblock on behalf of @MetaMask/design-system-engineers

bfullam committed:
feat: add default tron token pairs

bfullam committed:
chore: account for multichain transactions with no priority fee

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
fix: don't display fees if both are non-fungible

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
chore: add tron to allowed bridge chain IDs

bfullam committed:
fix: gate tron scope import with its own fence

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
Merge branch 'main' into swaps-tron-integration

GeorgeGkas committed:
feat: remove Tron non tradable assets from token selector

bfullam committed:
Merge branch 'main' into swaps-tron-integration

GeorgeGkas committed:
chore: Merge branch 'swaps-tron-integration' into swaps-3419-filter-out-non-tradable-assets

GeorgeGkas committed:
refactor: optimize

GeorgeGkas committed:
refactor: use existing structs

GeorgeGkas committed:
test: increase coverage to 100%

bfullam committed:
feat: remove Tron non tradable assets from token selector (#22471)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Tron Snap currently treats 'energy' and 'bandwidth' as assets, which
causes them to appear in asset lists, including the Swaps asset pickers.
Since these are not tradeable assets, we need to implement filtering on
the asset pickers to exclude them from the UI. We ensure that 'energy'
and 'bandwidth' are not displayed in asset pickers where tradeable
assets are expected.

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: Remove non-tradable assets from asset picker

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/SWAPS-3419

## **Manual testing steps**

```gherkin
- Ensure `Energy`, `Bandwidth` and `Max bandwidth` tokens are not rendered in asset picker.
- Ensure that all tokens excepts the one mentioned above are still there.
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Filters Tron resource tokens (Energy, Bandwidth, Max Bandwidth) from
bridge token lists and adds comprehensive tests, including non-EVM
address normalization and edge cases.
> 
> - **Bridge Utils**:
> - Add `isTradableToken` utility to exclude Tron resource tokens
(`Energy`, `Bandwidth`, `Max Bandwidth`) on `TrxScope.Mainnet`.
> - **Hook (`useTokens`)**:
> - Filter non-tradable tokens via `isTradableToken` when building
`tokensWithoutBalance`, `allTokens`, and `tokensToRender`.
> - **Tests**:
>   - Expand `useTokens.test.ts` to cover:
> - Non-EVM address normalization via `formatAddressToAssetId` and error
on empty normalization.
> - Filtering of Tron non-tradable tokens across `tokensWithBalance`,
`topTokens`, and `remainingTokens`.
> - Deduplication/combination logic and handling of undefined/null
arrays and exclusions.
> - Add unit tests for `isTradableToken` with Tron and non-Tron
scenarios.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
38c4cfc9e4e7f8f22de86268427e7585eed62fd8. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

bfullam committed:
chore: bump tron snap

bfullam committed:
Merge branch 'main' into swaps-tron-integration

bfullam committed:
chore: bump controllers

bfullam committed:
Merge branch 'swaps-tron-integration' of https://github.com/MetaMask/metamask-mobile into swaps-tron-integration
